src/clj/rems/api/licenses.clj

```diff
@@ -39,4 +61,34 @@
       :roles #{:owner}
       :body [command CreateLicenseCommand]
       :return CreateLicenseResponse
-      (ok (licenses/create-license! command)))))
+      (ok (licenses/create-license! command)))
+
+    (POST "/add_attachment" []
+      :summary "Add an attachment file that will be used in a license"
+      :roles #{:owner}
+      :multipart-params [file :- upload/TempFileUpload]
+      :middleware [upload/wrap-multipart-params]
+      :return AttachmentMetadata
+      (check-attachment-content-type (:content-type file))
+      (ok (licenses/create-license-attachment! file (util/getx-user-id))))
+
+    (POST "/remove_attachment" []
+      :summary "Remove an attachment that could have been used in a license."
+      :roles #{:owner}
+      :query-params [attachment-id :- (describe s/Int "attachment id")]
+      :return SuccessResponse
+      (if (some? (licenses/remove-license-attachment! attachment-id))
+        (ok {:success true})
+        (ok {:success false})))
+
+    (GET "/attachments/:attachment-id" []
+      :summary "Get a license's attachment"
+      :roles #{:owner}
+      :path-params [attachment-id :- (describe s/Int "attachment id")]
+      (if-let [attachment (db/get-license-attachment {:attachmentId attachment-id})]
+        (do (check-attachment-content-type (:type attachment))
+            (-> (:data attachment)
+                (java.io.ByteArrayInputStream.)
```

Comment:
Not sure but can the stream stay open if something unexpected happens?

Reply:
The attachment is entirely read into memory and this stream is to that memory buffer so there is nothing special that can leak in the case of an exception. Only regular memory handled by the Garbage Collector.